ignite/handlers/checkpoint.py

```diff
@@ -356,13 +356,12 @@ def __call__(self, engine: Engine) -> None:
                 "priority": priority,
             }
 
-            filename_already_exists = any(item.filename == filename for item in self._saved)
+            saved = [item for item in self._saved if item.filename != filename]
 
-            if filename_already_exists:
+            if self._saved != saved:
                 if isinstance(self.save_handler, BaseSaveHandler):
                     self.save_handler.remove(filename)
-                # list is purged
-                self._saved = [item for item in self._saved if item.filename != filename]
+                self._saved = saved
```

Comment:
Thanks Sylvain! I'm a bit bothered by the fact that we redefine `self._saved` every time if `filename_already_exists`. I was thinking if we could get the Item that has existing filename and then call remove on the list of `self._saved` ? What do you think ?

Reply:
I agree with you.